internal/model/message.go

```diff
@@ -0,0 +1,39 @@
+package model
+
+import (
+	"gorm.io/gorm"
+	"time"
+)
+
+// By is the type of entity that sent the message
+type By string
+
+const (
+	ByGuest By = "guest"
+	ByBot   By = "bot"
+	ByStaff By = "staff"
+)
+
+type Message struct {
+	gorm.Model
+	TelegramMessageId int64
+	By                By
+	MessageBody       string
+	Timestamp         time.Time
+	HotelStaffId      *uint
+	HotelStaff        *User
```

Comment:
again, not familiar w gorm but is this a 'belongs-to' hotelstaff? will this be problematic since not all messages are from hotel staff? (if its a 'has-one' ignore me)

Reply:
haven't used gorm much myself but the two testcases show:
"can create message from bot" - create message w/ no user
"can create chat without booking id" - create chat w/ no booking

Reply:
See leng is right that pointer fields means the field is nullable: https://gorm.io/docs/models.html#Declaring-Models

The other benefit for using pointer even for fields that is not nullable is deciding to lazy/eager preload these fields: https://gorm.io/docs/preload.html

By default, when Go instantiate a struct, all fields will be populated with their respective "0" values. If a model struct is composed of other structs, then Go needs to waste time to allocate memory for these field structs even when we do not need it.